### ISSUE TITLE
H.264 Nal Parsing

### ISSUE THREAD

**Issue by web2098** [paullouisageneau/libdatachannel#751]
Hi, while investigating how to depacketize an H.264 stream I used the code at StartSequenceMatchSucc to retrieve all the fragmented NAL units. While using this on an H.264 stream I had generated, I noticed it was missing a fragmented NAL unit. The reason the NAL unit was missed was because the StartSequenceMatchSucc doesn't handle the case where the start code (0x000001, 0x00000001) is preceded by a zero. While this might be specific to my case of using the code to undo the fragmentation. I have seen H.264 data that can contain multiple slices of nal data, so it seems plausible that this same issue could effect the H.264 packetizer code.

Basically what is happening is if you have a byte sequence of 0x00, 0x00, 0x00, 0x0, 0x01
It returns these states : NUSM_firstZero, NUSM_secondZero, NUSM_thirdZero, NUSM_noMatch

My Quick modification to the code was to do continue to return NUSM_thirdZero(or NUSM_secondZero) while the byte is zero. This results in the this sequence of states: NUSM_firstZero, NUSM_secondZero, NUSM_thirdZero, NUSM_thirdZero, NUSM_longMatch

**Comment by paullouisageneau** [comment]
Good catch, thank you! Would you mind opening a PR with your fix?
